Address scenarios with close page-media relations

splitbrain opened the pull request:
Some people use media files in close relation with the pages. Eg. media is rarely reused across different pages and rarely across namespaces. Usually media files are located in namespaces named exactly like the namespace of the page they are used in. In this scenario it is desirable to move media together with the page. (See #222, #269, #273 and possibly others).

This PR addresses this scenario in two ways:

1. The rename dialog has a new option called "Move referenced media files along with the page". When using this option all media that resides in the same namespace (named the same) as the page *and* is used in the page will be moved with the page if the rename operation assigns it a new namespace. See 64e370eb7e8064ba13451de082bd64a81c90d26a for details
2. A full rewrite of the Tree Manager allows users to view a tree that is a representation of page and media file trees merged into one. This is similar to the view we use in the ACL manager. This makes it easier in the above scenario to rename/move whole hierarchies but keeping pages and media files together. Most of the work is done in 2ddb0d5a89cfea21f4e9e4e839d3745daae7fd52

splitbrain committed:
optionally move media with page rename  #222

A new checkbox allows to move referenced page media along with a page
rename.

This happens only if

* the checkbox is ticked
* the page actually moves to a new namespace
* the page was not in the root namespace
* the referenced media is in the same namespace as the page

# Conflicts:
#	action/rename.php

splitbrain committed:
Rewrite of the tree manager

This rewrite drops all dependencies on jQuery for the tree manager and
uses modern JavaScript for the drag'n'drop operations. It introduces the
posibility to view the tree as a merged tree of page and media
namespaces (similar to what we do in the ACL manager). This merged view
is often advantageous in wikis where both trees are very similar. A
default view can be set in the configuration but users can switch
between the views on a case by case basis.

This does not change any functionality in any of the other scripts or in
the way how the move is executed. It's a GUI refresh only.

Other parts of the plugin could use a GUI refresh as well (eg. use the
same SVG based icons and drop jQuery dependencies).

splitbrain committed:
remove changed language string

The duplicate message now displays a full ID including the namespace,
so the string had to be adjusted. This removes it from all language
files to have translators retranslate it.

 Conflicts:
	lang/cs/lang.php
	lang/zh/lang.php

splitbrain committed:
better contrast between pages and media files

splitbrain committed:
remove accidentally checked in backup file

splitbrain committed:
drop obsolete code in tree component

splitbrain committed:
open the tree at the current page's position

splitbrain committed:
fix move into self check

We let the browser throw an error instead of trying to figure it out
ourself.

splitbrain committed:
fix dragging to root, we need an element for that

splitbrain committed:
fix duplicate checking

It's fine to have a page and a namespace named the same. Or a media file
and a page named the same.

michitux commented:
Thank you very much for this pull request with this nice feature! I've just tested this a tiny bit and while it looks awesome, I noticed a couple of small regressions compared to the current UI:

* When accepting the rename input without any change, an error regarding a duplicate item is displayed, even though the "duplicate" is the item itself. 
* When changing the name of an item back to the original name, the highlighting of the change doesn't disappear. This isn't just a UI issue, it actually tries to rename the page to itself which then fails with an error while before applying this PR, it correctly says that nothing has been changed.

![Screenshot_20250610_224153_Firefox.jpg](https://github.com/user-attachments/assets/ef8b22e7-0ab3-4335-86cb-ec4161ff8d08)

michitux commented:
Sorry for the confusing pushes. I first wanted to merge this directly, manually due to the merge conflicts but then I saw the problems mentioned in the previous comment. I then tried pushing the merge to this branch but somehow GitHub then displayed all commits from the master branch in this PR (maybe because the parents were in the wrong order as I merged the media branch into the master branch). To clean this up again I've now force-pushed it to the previous state with the merge conflict.

splitbrain commented:
I rebased the branch on current master. I will have a look at the issues later. Thanks for having a look!

splitbrain committed:
guard against renaming to empty or same ID

splitbrain committed:
fix issues with self-moves

* avoid moves to the same destination in admin interface
* avoid renames to self in tree manager
* avoid moves where items have been moved back to the original
  destination in tree manager

Note: all of this is mostly cosmetics. Self-moves are also detected
during plan execution, trigger an error and can be skipped.

splitbrain commented:
The issues should be addressed now.
